root/io/hadd/merge_check_output.C

```diff
@@ -0,0 +1,82 @@
+#include <ROOT/RLogger.hxx>
+#include <ROOT/RNTupleModel.hxx>
+#include <ROOT/RNTupleReader.hxx>
+#include <ROOT/RNTupleWriter.hxx>
+#include <ROOT/RPageStorageFile.hxx>
+#include <Compression.h>
+
+#include <TSystem.h>
+
+#include <string>
+#include <utility>
+#include <sstream>
+#include <cstdint>
+#include <cstring>
+
+// Copy-pasted from gtest-internal.h
+namespace {
+const uint32_t kMaxUlps = 4;
+const size_t kBitCount = 8 * sizeof(float);
+const uint32_t kSignBitMask = static_cast<uint32_t>(1) << (kBitCount - 1);
+
+uint32_t SignAndMagnitudeToBiased(const uint32_t &sam)
+{
+   if (kSignBitMask & sam) {
+      // sam represents a negative number.
+      return ~sam + 1;
+   } else {
+      // sam represents a positive number.
+      return kSignBitMask | sam;
+   }
+}
+
+uint32_t DistanceBetweenSignAndMagnitudeNumbers(const uint32_t &sam1, const uint32_t &sam2)
+{
+   const uint32_t biased1 = SignAndMagnitudeToBiased(sam1);
+   const uint32_t biased2 = SignAndMagnitudeToBiased(sam2);
+   return (biased1 >= biased2) ? (biased1 - biased2) : (biased2 - biased1);
+}
+
+bool FloatAlmostEquals(float a, float b)
+{
+   uint32_t ia, ib;
+   memcpy(&ia, &a, sizeof(float));
+   memcpy(&ib, &b, sizeof(float));
+   return DistanceBetweenSignAndMagnitudeNumbers(ia, ib) <= kMaxUlps;
+}
+} // namespace
+
+int merge_check_output(int expectedCompression, const char *fnameOut, const char *fnameIn1,
```

Comment:
The expected compression is unused.

Reply:
Copy-paste error, thanks for catching it!